test/unit/gcs.test.js

```diff
@@ -0,0 +1,64 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+// @flow
+
+import { Writable } from 'stream';
+
+import { Storage as MockStorage } from '../../__mocks__/@google-cloud/storage';
+import { Bucket } from '@google-cloud/storage';
+
+import { create as gcsStorageCreate } from '../../src/logic/gcs';
+
+beforeEach(() => MockStorage.cleanUp());
+afterEach(() => MockStorage.cleanUp());
+
+describe('logic/gcs', () => {
+  it('creates a mock service when asked for', async () => {
+    jest.spyOn(Bucket.prototype, 'exists');
+    jest.spyOn(Bucket.prototype, 'file');
+    const storage = gcsStorageCreate({
+      googleAuthenticationFilePath: 'MOCKED',
+      gcsBucket: '',
+    });
+
+    // This call shouldn't throw nor timeout.
+    await storage.ping();
+    expect(Bucket.prototype.exists).not.toHaveBeenCalled();
+
+    expect(storage.getWriteStreamForFile('filepath')).toBeInstanceOf(Writable);
+    expect(Bucket.prototype.file).not.toHaveBeenCalled();
+  });
+
+  it('calls the real service when passed no configuration file', async () => {
+    jest.spyOn(Bucket.prototype, 'exists');
+    jest.spyOn(Bucket.prototype, 'file');
+
+    const storage = gcsStorageCreate({
+      googleAuthenticationFilePath: '',
+      gcsBucket: 'profile-store',
+    });
+
+    await storage.ping();
+    expect(Bucket.prototype.exists).toHaveBeenCalled();
+
+    expect(storage.getWriteStreamForFile('filepath')).toBeInstanceOf(Writable);
+    expect(Bucket.prototype.file).toHaveBeenCalledWith('filepath');
+  });
+
+  it('calls the real service when passed a configuration file', async () => {
```

Comment:
Um, does it really?

Reply:
Yes, I think so, why not?
Basically I want to distinguish between using "MOCKED" or another value.